runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -892,23 +901,109 @@ void J9::RecognizedCallTransformer::processUnsafeAtomicCall(TR::TreeTop* treetop
             arrayAccessTreeTop->insertTreeTopsAfterMe(TR::TreeTop::create(comp(), storeNode));
             }
 
+         isObjectNullNode->setBranchDestination(treetop->getEnclosingBlock()->getEntry());
+         cfg->addEdge(TR::CFGEdge::createEdge(isObjectNullTreeTop->getEnclosingBlock(), treetop->getEnclosingBlock(), comp()->trMemory()));
+
          if (enableTrace)
             traceMsg(comp(), "Created array access helper block_%d that loads dataAddr pointer from array object address\n", arrayAccessBlock->getNumber());
          }
-   #endif /* J9VM_GC_SPARSE_HEAP_ALLOCATION */
+      else
+#endif /* J9VM_GC_SPARSE_HEAP_ALLOCATION */
+      if (!isNotStaticField)
+         {
+         /* Example of the atomic method helper
+         * n92n  treetop
+         * n93n    icall  <atomicFetchAndAdd>
+         * n94n      aladd
+         * n95n        aload  <temp slot 5>
+         * n96n        lload  <temp slot 3>
+         * n97n      iload  <temp slot 4>
+         */
+         // Create another helper call that loads from ramStatics
```

Comment:
at least, this example seemed contradictory to the setting of `isNotStaticField` (i.e.  definitely `true` if it is an atomic call ... this example is under `isNotStaticField` being `false`). overdoing something here? plus, it sounded like an existing bug for gencon anyway? (previously, it needs to handle ramStatics too).

Reply:
That part is moved not added, I guess its confusing given that the other code between the if-condition and that comment have moved out to a shared path. But the comment only talks about the subsequent section of code.
https://github.com/eclipse-openj9/openj9/blob/v0.49.0-release/runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp#L774-L808
I'll try to clean-up the code more making it less-confusing.

Reply:
Updated comments, ready for review